pkg/features/kube_features.go

```diff
@@ -570,6 +570,12 @@ const (
 	//
 	// Enables usage of any object for volume data source in PVCs
 	AnyVolumeDataSource featuregate.Feature = "AnyVolumeDataSource"
+
+	// owner: @nckturner
+	// alpha: v1.18
+	//
+	// Enables a feature to use external container image registry credential provider plugins.
+	ExternalRegistryCredentialProviders featuregate.Feature = "ExternalRegistryCredentialProviders"
```

Comment:
Should we define this feature publicly rather than privately? I realize that this is primarily intended to be used by the Kubelet, but it will be partnered with a side car, which we may want reading the value. This will also be true of any Operator which may function on such side cars. Currently the public feature gates seem to be in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/features/kube_features.go